### PR TITLE
fix: resolved 'Workspace not found' issue in devcontainer setup

### PR DESCRIPTION
**Pull request by Darshan808**
#### On setting up CircuitVerse through dev container, the error `Workspace not found` was encountered.
Screenshot (104)

#### To resolve this, I added a line in the docker-compose file of the devcontainer to mount the CircuitVerse Project folder into the workspace. This change ensures that developers and contributors can now use the devcontainer for their development environment setup without encountering this error.
Screenshot (109)

## Summary by CodeRabbit

- **New Features**
	- Introduced a volume mapping in the development container configuration, allowing seamless file access between the host and the container for improved development workflow.

### CODE REVIEW

File: .devcontainer/docker-compose.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ services:
       CIRCUITVERSE_USE_SOLR: "false"
       DOCKER_ENVIRONMENT: "true"
       DEV_CONTAINER: "true"
+    volumes:
+    - ..:/workspaces/CircuitVerse:cached
```
Comment (tanmoysrt) [inline .devcontainer/docker-compose.yml:23]: `:cached` probably has been deprecated after introducing of FUSE file sharing in docker desktop.

Ref: https://github.com/docker/for-mac/issues/5402 

Can you try out to remove `:cached` flag, if it still works ?

**Review by tanmoysrt — CHANGES_REQUESTED** [review]

File: .devcontainer/docker-compose.yml (hunk 1 above)
Comment (Darshan808) [reply .devcontainer/docker-compose.yml]: @tanmoysrt I force-pushed the ammend commit to this branch. Do i need to create a New PR or are the changes visible within this PR ?

**Review by tanmoysrt — APPROVED** [review]

**Comment by tachyons** [comment]
@Darshan808 Could you investigate why the checks aren't green

**Comment by Darshan808** [comment]
@tachyons
Thank you for your feedback! I have checked and it seems that all the necessary checks are green. Could you please let me know if there is a specific check you'd like me to investigate further? I'd be happy to look into it.

**Comment by Darshan808** [comment]
@tachyons  I noticed that one of the checks isn't passing. After consulting with Tanmoy, I found that the issue was related to the CI configuration. I've addressed this in #5080 

Once #5080 is merged, the CI tests should pass smoothly, and we can proceed with merging this one as well.

**Comment by tachyons** [comment]
@Darshan808 Great job

**Pull request merged**